src/sql/types.ts

```diff
@@ -7,4 +7,5 @@ export interface SQLQuery extends DataQuery {
   rawSQL: string;
   format?: number;
   fillMode?: { mode: FillValueOptions; value?: number };
+  meta?: { queryFlow?: 'async' | 'sync' };
```

Comment:
The frontend sends this to the backend to determine whether to use the sync (existing) query flow or the async query flow. The frontend sets this field based on a feature toggle. This is sent from the frontend because external plugins cannot access Grafana backend feature toggles

Reply:
is it odd that this would be saved in the query? How would we update old queries? if we have to revert after releasing would that be ok if queries are saved with this field?

I wonder if we could somehow right our frontend code to make the query not knowing if it is sync or async and then make different choices based on the response from the backend instead?

Reply:
This should not be saved in the query. The frontend adds this to the query right before it is executed. So old queries wouldn't need to know about this field. If I'm not mistaken, this only gets saved in the query if it's called with `onChange`, which isn't called between the time this field is added and when we execute the query. Here's where it's added in the redshift datasource https://github.com/grafana/redshift-datasource/pull/177/files#diff-804a5756e358074cb0ba27b1932cca77d39b90694fc1d4d122f46348c8653259R113.

The backend would need to decide when to run a query sync or async. Right now, we decide that based on a feature toggle. But since external plugins don't have access to the Grafana backend feature toggles, we're reading it from the frontend and making the decision there.

Reply:
Ah interesting. I wonder if it would somehow make sense to move this out of the query object then? Like it's own separate state concept? Open to whatever you think is best though!

Reply:
Yea that's a good idea. I've moved this to an interface in the DatasourceWithAsyncBackend.ts file.